src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -71,105 +80,134 @@ const NavHeader = (props) => {
+                paddingTop: insets.top,
+                paddingRight: dimen.normalizeDimen(20),
+                zIndex: 2,
+                position: 'absolute',
+                right: 0
+              })
+            }>
+            {!isUserFollow && isHeaderHide ? (
+              <ButtonFollow handleSetFollow={handleFollowTopic} />
+            ) : (
+              <TouchableOpacity onPress={onShareCommunity} style={styles.shareIconStyle}>
+                <ShareIconCircle color="black" width={32} height={32} />
+              </TouchableOpacity>
+            )}
+          </View>
         </View>
-      </View>
-      <Animated.View style={[styles.headerContainer]}>
-        {!isHeaderHide ? (
+        {!hasSearch && (
           <>
-            <Animated.View style={{opacity: opacityHeaderAnimation}}>
-              <FastImage
-                source={topicDetail?.icon_path ? {uri: topicDetail?.icon_path} : TopicDefaultIcon}
-                style={styles.image}
-              />
-            </Animated.View>
-            <View style={styles.containerAction}>
+            <View style={[styles.headerContainer]}>
               <Animated.View style={{opacity: opacityHeaderAnimation}}>
-                {isFollow ? (
-                  <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
-                ) : (
-                  <ButtonFollow handleSetFollow={handleFollowTopic} />
-                )}
+                <FastImage
+                  source={
+                    initialData?.channelPicutre
+                      ? {uri: initialData?.channelPicutre}
+                      : topicDetail?.icon_path
+                      ? {uri: topicDetail?.icon_path}
+                      : TopicDefaultIcon
+                  }
+                  style={styles.image}
+                />
               </Animated.View>
+              <View style={styles.containerAction}>
+                <Animated.View style={{opacity: opacityHeaderAnimation}}>
+                  {isUserFollow === undefined && isLoading ? (
+                    <Shimmer width={normalize(100)} height={normalize(36)} />
+                  ) : isUserFollow ? (
+                    <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
+                  ) : (
+                    <ButtonFollow handleSetFollow={handleFollowTopic} />
+                  )}
+                </Animated.View>
+              </View>
             </View>
+            <Animated.View
+              style={[
+                styles.domain(isHeaderHide),
+                {bottom: normalize(20), opacity: opacityHeaderAnimation}
+              ]}>
+              <TopicDomainHeader {...props} />
+            </Animated.View>
           </>
-        ) : null}
-      </Animated.View>
-      <View
-        onLayout={onDomainLayout}
-        style={[styles.domain(isHeaderHide), {bottom: getBottomPostition()}]}>
-        <TopicDomainHeader {...props} />
+        )}
       </View>
 
       {hasSearch && (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This is a significant refactor of the component's render method. The changes seem to be primarily focused on improving the visual appearance and user experience. However, there are some potential issues:

1. On line 101-129, you're conditionally rendering a shimmer effect based on the `isLoading` prop. This is a good practice for indicating loading state. However, please ensure that `isLoading` is being correctly updated in the parent component to avoid an infinite loading state.

2. On line 134-138 and 150-159, you're using absolute positioning with zIndex. While this can work, it can also lead to layout issues especially on different screen sizes or orientations. Consider using flexbox for layout where possible as it's more flexible and less prone to such issues.

3. On line 161-167, you're conditionally rendering either a follow button or a share icon based on whether the user is following the topic. This might confuse users as the UI changes based on their follow state. It might be better to always show both options but disable the follow button if the user is already following.

4. On line 187-193, you're conditionally rendering a shimmer effect, a follow button, or a following button based on the `isLoading` and `isUserFollow` props. This is a good practice for indicating loading state and the current follow state. However, please ensure that these props are being correctly updated to avoid incorrect UI states.

5. On line 208-210, you're conditionally rendering a search component based on the `hasSearch` prop. This is fine if you want to optionally show a search feature. However, please ensure that `hasSearch` is being correctly updated in the parent component.



<!-- This is an auto-generated comment by OSS CodeRabbit -->